bind/gengo.go

```diff
@@ -752,6 +752,97 @@ func (g *goGen) genType(sym *symbol) {
 		g.Printf("}\n\n")
 	}
 
+	if sym.isMap() {
+		ktyp := sym.GoType().Underlying().(*types.Map).Key()
+		ksym := g.pkg.syms.symtype(ktyp)
+		etyp := sym.GoType().Underlying().(*types.Map).Elem()
+		esym := g.pkg.syms.symtype(etyp)
+
+		if ksym == nil {
+			panic(fmt.Errorf("gopy: could not retrieve key type of %v", sym))
+		}
+
+		if esym == nil {
+			panic(fmt.Errorf("gopy: could not retrieve element type of %v", sym))
+		}
+
+		g.Printf("//export cgo_func_%[1]s_set\n", sym.id)
+		g.Printf("func cgo_func_%[1]s_set(self %[2]s, key %[3]s, value %[4]s) {\n",
+			sym.id,
+			sym.cgoname,
+			ksym.cgotypename(),
+			esym.cgotypename(),
+		)
+		g.Indent()
+		g.Printf("m := (*%[1]s)(unsafe.Pointer(self))\n", sym.gofmt())
+		g.Printf("if *m == nil {\n")
+		g.Indent()
+		g.Printf("*m = make(map[%[1]s]%[2]s)\n", ksym.cgotypename(), esym.cgotypename())
```

Comment:
@sbinet 
Hmm, The reason why I do that is when we converted from a Python dict into the Go map.
It will call _XXXX_new() to get new map object and then it tries to insert data into a new map object from Python but it failed with 
`assignment to entry in nil map.`